spring-cloud-kubernetes-config/src/main/java/org/springframework/cloud/kubernetes/config/reload/EventBasedConfigurationChangeDetector.java

```diff
@@ -66,7 +66,8 @@ public EventBasedConfigurationChangeDetector(AbstractEnvironment environment,
 	public void watch() {
 		boolean activated = false;
 
-		if (this.properties.isMonitoringConfigMaps()) {
+		if (this.properties.isMonitoringConfigMaps()
+				&& this.configMapPropertySourceLocator != null) {
```

Comment:
`null` checks wont be necessary if we are only creating these if the bean is present

Reply:
I put them in just because it is an either or. So you could have one or the other, and both watchers are being registered under the same changeDetector.